torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -411,6 +413,7 @@ void RequestCallbackImpl::processPythonRRefFetchCall(
       Message m =
           PythonRRefFetchRet(std::move(*result).toIValues()).toMessage();
       m.setId(messageId);
+      rref->waitAllDevices();
```

Comment:
this is the owner RRef, and after accessing the value from this RRef, we know that the `rpc.remote` request has already recorded CUDAEvents on this RRef, so we can access those events here. Then, we need to do two things:

1. use the `ctx->getStream(device)` API to grab a stream for each device this owner RRef uses. 
2. use owner RRef's CUDAEvents to block the streams allocated in (1). 

The reason that we need to block those streams instead of current streams is because, 1) later when tensorpipe sends out response, it will use the same ctx to get streams, and we need to make sure that comm ops inserted in those streams will wait for the pending computations inserted in the `rpc.remote` request. So one solution is to use the ctx as a proxy, where the `processFetch` add streams here and tensorpipe write reuses those streams; 2) we don't want to mess up with current/default streams, unless those streams are set by us, otherwise, the internal logic might interfere with user logic in the user main thread.